Objects/longobject.c

```diff
@@ -5421,6 +5432,59 @@ int_bit_length_impl(PyObject *self)
     return NULL;
 }
 
+/*[clinic input]
+int.bit_count
+
+Number of ones in the binary representation of self.
+
+>>> bin(13)
+'0b1101'
+>>> (13).bit_count()
+3
+[clinic start generated code]*/
+
+static PyObject *
+int_bit_count_impl(PyObject *self)
+/*[clinic end generated code: output=2e571970daf1e5c3 input=a428900d3e39a606]*/
+{
+    Py_ssize_t ndigits, i, bit_count = 0;
+    PyLongObject *result, *x, *y;
+
+    assert(self != NULL);
+    assert(PyLong_Check(self));
+
+    ndigits = Py_ABS(Py_SIZE(self));
+
+    for (i = 0; i < ndigits && i < PY_SSIZE_T_MAX/PyLong_SHIFT; i++) {
+        bit_count += popcount_digit(((PyLongObject *)self)->ob_digit[i]);
```

Comment:
popcount_digit could return a uint32_t, and bit_count could be a size_t rather than Py_ssize_t.

Reply:
I think this is fine as it is. We're not realistically ever going to hit the `Py_SSIZE_T_MAX/PyLong_SHIFT` limit.